src/main/java/com/exasol/adapter/document/documentfetcher/files/SegmentMatcher.java

```diff
@@ -0,0 +1,55 @@
+package com.exasol.adapter.document.documentfetcher.files;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * This class separates a set of files into multiple segments.
+ * 
+ * <p>
+ * This matcher is used on different parallel runs to partition the files. It uses a hash of the file name to determine
+ * the partition of the file. Due to that implementation the segments are not guaranteed to have exactly the same size.
+ * The reason for this implementation is, that it works independent of the order in that the files are matched.
+ * </p>
+ */
+class SegmentMatcher {
```

Comment:
Instead of using SHA256 digest, would using the file name hash code work?

```
fileName.hashCode() % getNumSegments()
```

Reply:
ok, that works too.

Reply:
:+1: